### ISSUE TITLE
Bug: scope-manager throws unhelpful `aaa` error message

### ISSUE THREAD

**Issue by abrahamguo** [typescript-eslint/typescript-eslint#11727]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

scope-manager

### Repro Code

```TypeScript
// Code example adapted from typescript-eslint.io/packages/scope-manager

import { analyze } from '@typescript-eslint/scope-manager';
import { parse } from '@typescript-eslint/typescript-estree';

analyze(parse(`const hello: string = 'world';`).body[0]);
```

### ESLint Config

```javascript
No ESLint config.
```

### tsconfig

```jsonc
{ "compilerOptions": { "module": "nodenext", "noEmit": true, "strict": true } }
```

### Expected Result

It should either:

- work at runtime, or
- fail at compile time with a TS error, AND fail at runtime with a *helpful* error message.

### Actual Result

- At compile time, `tsc` reports no errors, indicating that this is a valid usage.
- At runtime, `@typescript-eslint/scope-manager` throws `aaa`:

```
node_modules/@typescript-eslint/scope-manager/dist/assert.js:7
        throw new Error(message);
              ^

Error: aaa
    at assert (node_modules/@typescript-eslint/scope-manager/dist/assert.js:7:15)
    at Referencer.currentScope (node_modules/@typescript-eslint/scope-manager/dist/referencer/Referencer.js:65:33)
    at Referencer.VariableDeclaration (node_modules/@typescript-eslint/scope-manager/dist/referencer/Referencer.js:516:20)
    at Referencer.visit (node_modules/@typescript-eslint/scope-manager/dist/referencer/VisitorBase.js:59:21)
    at analyze (node_modules/@typescript-eslint/scope-manager/dist/analyze.js:39:16)
    at index.ts:4:1
```

### Additional Info

Looks like this error message has been present in the source code since the introduction of `@typescript-eslint/scope-manager` by @bradzacher in #1939.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/scope-manager`     | `8.46.2` |
| `@typescript-eslint/typescript-estree` | `8.46.2` |
| `TypeScript`                           | `5.9.3` |
| `node`                                 | `25.0.0` |

**Comment by bradzacher** [comment]
Technically there are probably numerous nodes from which you could start the analysis.

But in general the expectation is that you pass a `Program`. So by passing not a program you've circumvented a bunch of setup.

This should probably be a type error and it should always expect a program node.

But also the docs to have an (IMO) clear and explicit example of how to call it:

https://typescript-eslint.io/packages/scope-manager#analyzetree-options

I.e. You haven't parsed with ranges either and that'll potentially cause a crash later too.
